### PR TITLE
chore: Remove all old style mocks

### PR DESCRIPTION
**Pull request by johncowen**
This PR:

- Removes all the old style 'fixture-like' static mocks, we no longer need to support both formats.
- Moves all our new mocks to `@/test-support`.
- Move the code used for 'serving' mocks files during development and 'mocking' mock files during testing over to `@/test-support`. I'm still unsure on the naming of things like `fakeApi` and `mocker`, at least these are all in one place in `@/test-support` now - so if the names change at any point we can just use import aliasing if need be. All in all though its just easier that these 'test-support' functions are in one place (not in a couple of different places)

The PR is quite large unfortunately, so I tried to split it into easier to digest commits, so looking at those might help review. The majority of the PR is setting up tests with the `mock` function to ensure that the test pass without relying on the old static mocks. As always, any questions please ask 👍 

(I may do some inline self-review comments shortly to help)

Closes https://github.com/kumahq/kuma-gui/issues/522

Signed-off-by: John Cowen <john.cowen@konghq.com>

### CODE REVIEW

File: src/app/App.spec.ts
Comment (johncowen) [inline src/app/App.spec.ts:78]: I removed the snapshot match here as the previous test setup was still loading in other API requests perfectly fine, and was creating all the donut charts etc. Really we are testing for whether the `Unable to reach the API` message is showing, so I changed it to test just for that.

File: src/services/development.ts
Comment (johncowen) [inline src/services/development.ts:105]: This is cruft left over from a previous change, you can't pass arguments to `constant`s

File: src/services/development.ts
Comment (johncowen) [inline src/services/development.ts:93]: All the old style mocks have gone now.

File: src/services/development.ts
Comment (johncowen) [inline src/services/development.ts:120]: This was pretty much just moved, token name changed, and I also added the extra `rest` argument (`fakeApi(env, rest, fs)`). You'll see the final result over in the green of the changeset.

Comment (johncowen) [inline src/services/development.ts:120]: This was pretty much just moved, token name changed, and I also added the extra `rest` argument (`fakeApi(env, rest, fs)`). You'll see the final result over in the green of the changeset.

File: src/utilities/helpers.ts
Comment (johncowen) [inline src/utilities/helpers.ts:143]: I wanted to double check this before complete removal.

This seems to check that the `total` stays the same throughout all pages of a paged request, and if it doesn't then an error is thrown.

This feels to me like it is actually expected behaviour that the total could change between requesting page one and page two of a request. Erroring if the total changes is probably not quite what we want to do.

I came across this as during testing. As our mocks no longer have state, it is quite possible that the total is different between requesting page one and page two (etc etc).

All in all I think deleting this is safe, and desirable, but thought I'd ask for a second opinion incase.

I'll be either deleting or uncommenting this before merge (probably deleting)

**Review by johncowen — COMMENTED** [review]
Self-review inline notes:

**Review by lahabana — APPROVED** [review]

**Pull request merged**
